packages/frontend/src/pages/Admin.tsx

```diff
@@ -154,6 +281,118 @@ function ReportedRatings() {
+                                onClick={resumeFromCursor}
+                                disabled={!resumeCursor.trim()}
+                                className="bg-blue-500 hover:bg-blue-600"
+                            >
+                                Resume
+                            </Button>
+                        </div>
+                        {auditProgress.nextCursor && (
+                            <div className="text-sm">
+                                <p className="text-gray-600 mb-1">Current cursor:</p>
+                                <code className="bg-gray-100 px-2 py-1 rounded text-xs break-all">
+                                    {auditProgress.nextCursor}
+                                </code>
+                                <button
+                                    type="button"
+                                    onClick={() => {
+                                        navigator.clipboard.writeText(
+                                            auditProgress.nextCursor || "",
+                                        );
+                                    }}
+                                    className="ml-2 text-blue-500 hover:text-blue-700 text-xs"
+                                >
+                                    Copy
+                                </button>
+                            </div>
+                        )}
+                    </div>
+                )}
+
+                {/* Progress Display */}
+                {(auditProgress.processedCount > 0 || auditProgress.isRunning) &&
+                    (() => {
+                        const progressPercent =
+                            auditProgress.totalProfessors > 0
+                                ? (auditProgress.processedCount / auditProgress.totalProfessors) *
+                                  100
+                                : 0;
+                        return (
+                            <div className="text-sm space-y-1">
+                                <div>
+                                    Progress: {auditProgress.processedCount} /{" "}
+                                    {auditProgress.totalProfessors} professors (
+                                    {Math.round(progressPercent)}
+                                    %)
+                                </div>
+                                <div>Duplicates Found: {auditProgress.duplicatesFound}</div>
+                                <div>Moderation Flagged: {auditProgress.moderationFlagged}</div>
+                                {auditProgress.isRunning && (
+                                    <div className="w-full bg-gray-200 rounded-full h-2">
+                                        <div
+                                            className="bg-blue-600 h-2 rounded-full transition-all duration-300"
+                                            style={{
+                                                // eslint-disable-next-line max-len
```

Comment:
Maybe another sign that this should be refactored. The line length requirments in this project are quite flexable